src/emu-i386/simx86/sigsegv.c

```diff
@@ -361,6 +340,24 @@ int e_emu_fault(struct sigcontext *scp)
 	return TryMemRef;
   }
 #endif
+
+  /* CONFIG_CPUSIM only: */
+  if (_trapno==0x0e) {
+	if (in_dpmi_emu) {
+		/* reflect DPMI page fault back to a DOSEMU crash or
+		   DPMI exception;
+		   vm86 faults will terminate DOSEMU via "return 0"
+		*/
+		TheCPU.err = EXCP0E_PAGE;
+		TheCPU.scp_err = _err;
+		TheCPU.cr2 = _cr2;
+		TheCPU.eip = P0 - LONG_CS;
+		fault_cnt--;
+		siglongjmp(jmp_env, 0);
```

Comment:
I don't think it can be easily removed: the interpreted code causes a SIGSEGV, and this exception needs to potentially be handled by a registered DPMI exception handler (see the tests in src/tests/). The alternative is to return to codegen-sim.c but that then needs to put in checks at every memory access (expensive for a very exceptional situation).